nixos/modules/services/display-managers/greetd.nix

```diff
@@ -102,12 +102,18 @@ in
 
     systemd.defaultUnit = "graphical.target";
 
+    # Create directories potentially required by supported greeters
+    # See https://github.com/NixOS/nixpkgs/issues/248323
+    systemd.tmpfiles.rules = [
+      "d '/var/cache/tuigreet' - greeter greeter - -"
```

Comment:
This doesn't account for `services.greetd.settings.user`

Reply:
I'm aware of this, but so does the rest of the module, see https://github.com/katexochen/nixpkgs/blob/3e4e76e6769bab6fa7f1f6f9d63967147e701ac8/nixos/modules/services/display-managers/greetd.nix#L111-L116